### PR TITLE
PHPStan (Level 2) on controllers/

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | PHPStan (Level 2) on controllers/
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Relative to #16471
| How to test?      | Github Actions are :green_circle: & @boubkerbribri tests are **QA :heavy_check_mark:**

**:notebook: BC Breaks**
* Remove static method `recurseCategoryForInclude` in `AdminProductsController`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26864)

### CODE REVIEW

File: controllers/admin/AdminCountriesController.php [hunk 1]
```diff
@@ -478,14 +479,6 @@ protected function processBulkAffectZone()
 
     protected function displayValidFields()
     {
-        /* The following translations are needed later - don't remove the comments!
```
Comment (matks) [inline controllers/admin/AdminCountriesController.php:481]: Well, this line tells you to NOT remove the comments 😛  so I think unless we are 100% sure it's OK to remove it, better keep it. We have some magic related to translation scanning that might need this.

**Review by matks — CHANGES_REQUESTED** [review]
CI fails

File: controllers/admin/AdminCountriesController.php (hunk 1 above)
Comment (Progi1984) [reply controllers/admin/AdminCountriesController.php:481]: :+1:

Comment (matks) [reply controllers/admin/AdminCountriesController.php:481]: Thanks for changing !

Maybe, to be sure, it works with `[]` we should run the catalog export process 🤔  ?

**Review by matks — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by Progi1984** [comment]
@boubkerbribri The :soccer: is on your side.

**Review by boubkerbribri — APPROVED** [review]
UI tests are all :green_circle: 
Thanks @Progi1984

**Comment by Progi1984** [comment]
Thanks @boubkerbribri

**Pull request merged**